syntax = "proto3";

package tutorial;

option java_package = "gen";
option java_outer_classname = "EventProtos";
option java_multiple_files = true;

service EventSubscription{
    rpc subscribeOnEvent (EventRequest) returns (stream Event);
    rpc connectToServer (Connect) returns (Connect);
}

message EventRequest {
    EventType eventType = 1;
}

message Event {
    string name = 1;
    EventType eventType = 2;
    string city = 3;
    int32 entranceFee = 4;
    int32 duration = 5;
    repeated Person organizer = 6;
}

enum EventType{
    CONCERT = 0;
    FESTIVAL = 1;
    EXHIBITION = 2;
    PARADE = 3;
}

message Person{
    string firstName = 1;
    string lastName = 2;
    string phoneNumber = 3;
}

message Connect{}
